### PR TITLE
Update destroy

### PR DESCRIPTION
**Pull request by ezesplit**
Metrics, events, and impressions are sent upon destroy
Main thread waits for posts to Split backend
at_exit registration calls destroy on interrupts
fixes clear not delegated in impressions_repository

### CODE REVIEW

**Comment by mmelograno** [comment]
@ezesplit @ignaciosplit is this ticket ready to review? There are some @Todo on the description.

**Comment by ignaciosplit** [comment]
@mmelograno Yes it's ready. I already removed the TODOs.

File: lib/splitclient-rb/cache/repositories/events/redis_repository.rb [hunk 1]
```diff
@@ -16,11 +16,23 @@ def add(key, traffic_type, event_type, time, value)
...
           end
 
-          def clear
-            @adapter.get_from_queue(namespace_key('.events'), EVENTS_SLICE).map do |e|
+          def get_events(number_of_events = 0)
+            @adapter.get_from_queue(namespace_key('.events'), number_of_events).map do |e|
               JSON.parse(e, symbolize_names: true)
             end
+          rescue StandardError => e
+            SplitIoClient.configuration.logger.error("Exception while clearing impressions cache: #{e}")
```
Comment (NicoZelaya) [inline lib/splitclient-rb/cache/repositories/events/redis_repository.rb]: Is the comment correct? Is this for impressions or events?

File: lib/splitclient-rb/cache/senders/events_sender.rb [hunk 2]
```diff
@@ -27,18 +27,25 @@ def call
...
 
-              sleep(SplitIoClient::Utilities.randomize_interval(SplitIoClient.configuration.events_push_rate))
+              SplitIoClient.configuration.logger.info('Posting events due to shutdown')
             end
           end
         end
 
-        def post_events
-          events_api.post(@events_repository.clear)
+        def post_events(fetch_all_impressions = true)
```
Comment (NicoZelaya) [inline lib/splitclient-rb/cache/senders/events_sender.rb]: shouldn't be "fetch_all_events" ?

**Pull request merged**